app/server.ts

```diff
@@ -0,0 +1,41 @@
+import * as Express from 'express';
+import * as BodyParser from 'body-parser';
+import * as Morgan from 'morgan';
+
+import { PORT } from './modules/util/config';
+import { BasePath as ProjectBasePath, projectRouter } from './modules/project';
```

Comment:
This means, your server is tightly coupled with your project entities, which is good way to start but you loose the separation of concerns with respect to this server class.

Think about it, a web server is an independent running unit, which can expect any sort of routes, provided they are mounted/attached to it.
* It should be able to be `unit tested`.
* It should be able to cater all your requirements for a web server.
* It should be configurable.

- [ ] Make server loosely coupled with application domain.
- [ ] Expect as many as configurations to spin up the server.